homeassistant/components/synology_dsm/camera.py

```diff
@@ -1,68 +1,87 @@
 """Support for Synology DSM cameras."""
+import logging
 from typing import Dict
 
 from synology_dsm.api.surveillance_station import SynoSurveillanceStation
-from synology_dsm.api.surveillance_station.camera import SynoCamera
+from synology_dsm.exceptions import SynologyDSMAPIErrorException
 
 from homeassistant.components.camera import SUPPORT_STREAM, Camera
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
-from . import SynoApi, SynologyDSMEntity
+from . import SynoApi, SynologyDSMCoordinatorEntity
 from .const import (
+    COORDINATOR_SURVEILLANCE,
     DOMAIN,
-    ENTITY_CLASS,
     ENTITY_ENABLE,
-    ENTITY_ICON,
     ENTITY_NAME,
-    ENTITY_UNIT,
     SYNO_API,
 )
 
+_LOGGER = logging.getLogger(__name__)
+
 
 async def async_setup_entry(
     hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
 ) -> None:
-    """Set up the Synology NAS binary sensor."""
+    """Set up the Synology NAS cameras."""
 
-    api = hass.data[DOMAIN][entry.unique_id][SYNO_API]
+    data = hass.data[DOMAIN][entry.unique_id]
+    api = data[SYNO_API]
 
     if SynoSurveillanceStation.CAMERA_API_KEY not in api.dsm.apis:
         return
 
-    surveillance_station = api.surveillance_station
-    await hass.async_add_executor_job(surveillance_station.update)
-    cameras = surveillance_station.get_all_cameras()
-    entities = [SynoDSMCamera(api, camera) for camera in cameras]
+    # initial data fetch
+    coordinator = data[COORDINATOR_SURVEILLANCE]
+    await coordinator.async_refresh()
 
-    async_add_entities(entities)
+    async_add_entities(
+        SynoDSMCamera(api, coordinator, camera_id)
+        for camera_id in coordinator.data["cameras"]
+    )
 
 
-class SynoDSMCamera(SynologyDSMEntity, Camera):
+class SynoDSMCamera(SynologyDSMCoordinatorEntity, Camera):
     """Representation a Synology camera."""
 
-    def __init__(self, api: SynoApi, camera: SynoCamera):
+    def __init__(
+        self, api: SynoApi, coordinator: DataUpdateCoordinator, camera_id: int
+    ):
         """Initialize a Synology camera."""
         super().__init__(
             api,
-            f"{SynoSurveillanceStation.CAMERA_API_KEY}:{camera.id}",
+            f"{SynoSurveillanceStation.CAMERA_API_KEY}:{camera_id}",
             {
-                ENTITY_NAME: camera.name,
-                ENTITY_CLASS: None,
-                ENTITY_ICON: None,
-                ENTITY_ENABLE: True,
-                ENTITY_UNIT: None,
```

Comment:
Let's keep these so we don't need a `.get` since defaults usually result in refactoring errors down the road.